### PR TITLE
Enable load after store CC tests in file collections subproject

### PR DESCRIPTION
**Pull request by wolfs**

### CODE REVIEW

File: subprojects/file-collections/src/integTest/groovy/org/gradle/api/file/FileCollectionLifecycleIntegrationTest.groovy [hunk 1]
```diff
@@ -124,6 +125,7 @@ class FileCollectionLifecycleIntegrationTest extends AbstractIntegrationSpec imp
         succeeds()
     }
 
+    @ToBeFixedForConfigurationCache(because = "Seems like finalizeValue() isn't stored in the CC")
```
Comment (wolfs) [inline subprojects/file-collections/src/integTest/groovy/org/gradle/api/file/FileCollectionLifecycleIntegrationTest.groovy:128]: Is there already an issue for this?

Comment (bamboo) [reply subprojects/file-collections/src/integTest/groovy/org/gradle/api/file/FileCollectionLifecycleIntegrationTest.groovy:128]: Fixed by #25499

File: subprojects/file-collections/src/integTest/groovy/org/gradle/api/file/FilePropertyLifecycleIntegrationTest.groovy [hunk 2]
```diff
@@ -207,6 +210,7 @@ task thing {
         output.count("prop = " + file("build/dir.out")) == 3
     }
 
+    @ToBeFixedForConfigurationCache(because = "Producer isn't attached to property used in before task")
```
[the remark this answers is not in the document]
Comment (bamboo) [reply subprojects/file-collections/src/integTest/groovy/org/gradle/api/file/FilePropertyLifecycleIntegrationTest.groovy:213]: As discussed, I don't think this is about the producer relationship but about `disallowUnsafeRead`, https://github.com/gradle/gradle/issues/25513

File: subprojects/file-collections/src/integTest/groovy/org/gradle/api/file/FilePropertyLifecycleIntegrationTest.groovy [hunk 3]
```diff
@@ -427,6 +432,7 @@ task thing {
         failureHasCause("Querying the mapped value of task ':producer' property 'output' before task ':producer' has completed is not supported")
     }
 
+    @ToBeFixedForConfigurationCache(because = "mapped task output property can be read before task completed")
```
[the remark this answers is not in the document]
Comment (bamboo) [reply subprojects/file-collections/src/integTest/groovy/org/gradle/api/file/FilePropertyLifecycleIntegrationTest.groovy:435]: This looks like https://github.com/gradle/gradle/issues/19252

File: subprojects/file-collections/src/integTest/groovy/org/gradle/api/file/FilePropertyLifecycleIntegrationTest.groovy [hunk 4]
```diff
@@ -484,6 +491,7 @@ task thing {
         failureHasCause("Querying the mapped value of task ':producer' property 'output' before task ':producer' has completed is not supported")
     }
 
+    @ToBeFixedForConfigurationCache(because = "Producer isn't attached to property used in before task")
```
Comment (bamboo) [inline subprojects/file-collections/src/integTest/groovy/org/gradle/api/file/FilePropertyLifecycleIntegrationTest.groovy:494]: I think this is `disallowUnsafeRead` again.

File: subprojects/file-collections/src/integTest/groovy/org/gradle/api/file/FilePropertyLifecycleIntegrationTest.groovy [hunk 5]
```diff
@@ -536,6 +544,7 @@ task thing {
         output.count("prop = " + file("build/dir.out")) == 1
     }
 
+    @ToBeFixedForConfigurationCache(because = "Producer isn't attached to property used in before task")
```
Comment (bamboo) [inline subprojects/file-collections/src/integTest/groovy/org/gradle/api/file/FilePropertyLifecycleIntegrationTest.groovy:547]: `disallowUnsafeRead`

File: subprojects/file-collections/src/integTest/groovy/org/gradle/api/file/FilePropertyLifecycleIntegrationTest.groovy [hunk 6]
```diff
@@ -90,6 +91,7 @@ class FilePropertyLifecycleIntegrationTest extends AbstractIntegrationSpec imple
         "@OutputDirectory" | _
     }
 
+    @ToBeFixedForConfigurationCache(because = "ends up being different properties after load")
```
[the remark this answers is not in the document]
Comment (bamboo) [reply subprojects/file-collections/src/integTest/groovy/org/gradle/api/file/FilePropertyLifecycleIntegrationTest.groovy:94]: I don't think so.

**Review by bamboo — APPROVED** [review]
Looks good!

File: subprojects/file-collections/src/integTest/groovy/org/gradle/api/file/FilePropertyLifecycleIntegrationTest.groovy (hunk 6 above)
[the remark this answers is not in the document]
Comment (wolfs) [reply subprojects/file-collections/src/integTest/groovy/org/gradle/api/file/FilePropertyLifecycleIntegrationTest.groovy:94]: I created https://github.com/gradle/gradle/issues/25516.

**Comment by wolfs** [comment]
@bot-gradle test and merge

**Review by Ghostman2591 — APPROVED** [review]

**Pull request merged**